### PR TITLE
Audit event messages filtering

### PR DESCRIPTION
**Pull request by nguyenfilip**
@mstead @awood This is the audit events filtering. See javadoc of the following: class EventFilter, ConfigProperties.AUDIT_FILTER*

### CODE REVIEW

**Comment by nguyenfilip** [comment]
Ups, missed a unit test in my commit. Amending.

File: server/src/main/java/org/candlepin/audit/EventFilter.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                        "Each filter auditing" +
+            "include/exclude must be in format TYPE-TARGET. For example MODIFIED-CONSUMER.");
+            }
+            try {
+                Type type = Enum.valueOf(Type.class, split[0].trim());
+                Target target = Enum.valueOf(Target.class, split[1].trim());
+                set.add(new EventTypeAndTarget(type, target));
+            }
+            catch (Exception ex) {
+                throw new IllegalArgumentException("Invalid filtering touple: " + item +
```
Comment (awood) [inline server/src/main/java/org/candlepin/audit/EventFilter.java]: Did you mean "tuple" here?

File: server/src/main/java/org/candlepin/audit/EventFilter.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        }
+
+        EventTypeAndTarget eventKey = new EventTypeAndTarget(event.getType(),
+                event.getTarget());
+
+        if (toNotFilter.contains(eventKey)) {
+            return false;
+        }
+
+
```
Comment (awood) [inline server/src/main/java/org/candlepin/audit/EventFilter.java]: Extra whitespace here.

File: server/src/main/java/org/candlepin/audit/EventSinkImpl.java [hunk 3]
```diff
@@ -159,11 +162,20 @@ protected ClientProducer getClientProducer() {
...
      * is not successful, rollback() must be called.
      *
+     * Events are filtered, meaning that some of them might not even get into HornetQ.
+     * Details about the filtering are documented in EventFilter class
+     *
      * HornetQ transaction actually manages the queue of events to be sent.
      */
     @Override
     public void queueEvent(Event event) {
+        if (eventFilter.shouldFilter(event)) {
```
Comment (awood) [inline server/src/main/java/org/candlepin/audit/EventSinkImpl.java]: Please drop this guard clause.

File: server/src/main/java/org/candlepin/audit/EventFilter.java (hunk 1 above)
Comment (nguyenfilip) [inline server/src/main/java/org/candlepin/audit/EventFilter.java]: Yes, amending.

File: server/src/main/java/org/candlepin/audit/EventFilter.java (hunk 2 above)
Comment (nguyenfilip) [inline server/src/main/java/org/candlepin/audit/EventFilter.java]: deleted

**Comment by awood** [comment]
:clap: Looks good.  I will merge when Jenkins finishes.

**Comment by nguyenfilip** [comment]
retest this please

**Comment by nguyenfilip** [comment]
retest this please

**Comment by nguyenfilip** [comment]
retest this please

**Comment by awood** [comment]
Pretty sure this is a legit unit test failure:
```
[junit] null
[junit] junit.framework.AssertionFailedError
[junit] 	at org.candlepin.controller.PoolManagerTest.expiredEntitlementEvent(PoolManagerTest.java:1532)
```

although I can't reproduce it locally. :frowning:

**Comment by vritant** [comment]
@awood @nguyenfilip I was hit by this issue too, fixed in #1001 's commit 7bf1b55f7268ef4a23edaf1cd311c68ead0be370

**Comment by nguyenfilip** [comment]
@awood yeah locally the unit tests also have been passing for me

**Comment by nguyenfilip** [comment]
retest this please

**Comment by nguyenfilip** [comment]
@awood the unit test seems to pass sometimes. Maybe Jenkins issue or timing issue

**Comment by vritant** [comment]
@nguyenfilip @awood   #1001 's commit 7bf1b55 will fix this non-deterministic test failure, so if that is the only issue with this PR, I'd say we are good.

**Comment by nguyenfilip** [comment]
Thank you @vritant  for the information.

**Pull request merged**